Source/WebCore/platform/network/ResourceResponseBase.cpp

```diff
@@ -87,14 +87,16 @@ ResourceResponseBase::ResourceResponseBase(std::optional<ResourceResponseBase::R
 ResourceResponseBase::CrossThreadData ResourceResponseBase::CrossThreadData::copy() const
```

Comment:
Seems like we should be calling this isolatedCopy() then? Maybe the copy() version that is cheaper is still valuable for call sites that don't need the isolated copying?

Reply:
CrossThreadData has no copy constructor by design.
The copy() version is dangerous, we should just not use it.
I prefer keeping copy() since isolatedCopy() would make think we could have a copy() version which we should not.

Reply:
But I think we need a copy() function that doesn't isolate copy. You new code is slower and as far as I can tell is not required in the common case. It is still unclear to me why you need to isolate copy at all. Maybe it would be clearer if I understand what the call site was doing.

Reply:
> But I think we need a copy() function that doesn't isolate copy.

The only use of copy() is in MemoryCacheStorage where it is unsafe.

Reply:
> The call site is doing.

The storage is keeping a list of record in memory, which contains ResourceResponses as CrossThreadData.
The storage is queried from time to time and answers with a ResourceResponse when there is a match.